viewing_party/party.py

```diff
@@ -0,0 +1,160 @@
+#Wave one
+def create_movie(title, genre, rating):
+    movie_dict = {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+    if title == None or genre == None or rating == None:
+        return None
+    else:
+        return movie_dict
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+  for movie_data in user_data["watchlist"]:
+    if movie_data ["title"] == title:
+      user_data["watched"].append(movie_data)
+      user_data["watchlist"].remove(movie_data)
+  return user_data 
+
+#Wave two
+def get_watched_avg_rating(user_data):
+    all_movie_ratings = []
+
+    for each_rating in user_data["watched"]:
+        all_movie_ratings.append(each_rating["rating"])
+
+    if len(all_movie_ratings) == 0:
+      return 0.0
+
+    average_rating_of_all = sum(all_movie_ratings)/len(all_movie_ratings)
+    return average_rating_of_all
+
+def get_most_watched_genre(user_data):
+  genre_map = {}
+
+  for genre in user_data["watched"]:
+    if genre["genre"] not in genre_map:
+        genre_map[genre["genre"]] = 1 
+    else:
+        genre_map[genre["genre"]] += 1
+
+  if len(user_data["watched"]) == 0:
+    return None
+
+  max_key = max(genre_map, key=genre_map.get)
+  return max_key
+
+#Wave Three
+def get_unique_watched(user_data):
+    #tyring out list/dictionary comprehension structure
+    user_watched_list = {y['title']: y['title'] for friends_watched in user_data['friends'] for y in friends_watched['watched']}
+    unique_data = [r for r in user_data['watched'] for k, v in r.items() if v not in user_watched_list.keys()]
+
+    return unique_data  
+
+def get_friends_unique_watched(user_data):
+    x = 1
+    user_watched_list = {watched['title']: watched['title'] for watched in user_data['watched']}
+    unique_data = [y for x in user_data['friends'] for y in x['watched'] if y.values() in user_watched_list.keys()]
+    return unique_data 
+
+def get_friends_unique_watched(user_data):
+    user_watched_list = {}
+    unique_data = []
+  
+    x = 1
+    for watched in user_data['watched']:
+        user_watched_list[watched['title']] = watched['title']
+
+    for each_friend in user_data['friends']:
+        if len(each_friend['watched']) > 0:
+            for x in each_friend['watched']:
+                for k, v in x.items():
+                    if v not in user_watched_list.keys():
```

Comment:
This could be rewritten a bit.

```suggestion
                    if v not in user_watched_list:
```